File: CS/ChangeDatabase.Module.Web/WebChangeDatabaseAuthenticationActiveDirectory.cs
using System;
using System.Web;
using DevExpress.ExpressApp.Security;
using DevExpress.Persistent.BaseImpl;

namespace ChangeDatabase.Module.Web {
    public class WebChangeDatabaseAuthenticationActiveDirectory : AuthenticationActiveDirectory<SimpleUser, ChangeDatabaseActiveDirectoryLogonParameters> {
        public WebChangeDatabaseAuthenticationActiveDirectory() {
            this.CreateUserAutomatically = true;
        }
        public override bool IsLogoffEnabled {
            get {
                return true;
            }
        }
        public override bool AskLogonParametersViaUI {
            get {
                string databaseName = HttpContext.Current.Request.Params[WebChangeDatabaseController.DatabaseParameterName];
                if(string.IsNullOrEmpty(databaseName)) {
                    return true;
                }
                return base.AskLogonParametersViaUI;
            }
        }
    }
}

File: VB/ChangeDatabase.Module.Web/WebChangeDatabaseAuthenticationActiveDirectory.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.ComponentModel
Imports System.Collections.Generic
Imports System.Diagnostics
Imports System.Text

Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Security
Imports System.Web
Imports DevExpress.ExpressApp.Web
Imports System.Web.Security

Namespace ChangeDatabase.Module.Web
	Public Class WebChangeDatabaseAuthenticationActiveDirectory
		Inherits AuthenticationActiveDirectory(Of DevExpress.Persistent.BaseImpl.SimpleUser, ChangeDatabaseActiveDirectoryLogonParameters)
		Public Sub New()
			Me.CreateUserAutomatically = True
		End Sub
		Public Overrides ReadOnly Property AskLogonParametersViaUI() As Boolean
			Get
				Dim databaseName As String = HttpContext.Current.Request.Params(WebChangeDatabaseController.DatabaseParameterName)
				If String.IsNullOrEmpty(databaseName) Then
					Return True
				End If
				Return MyBase.AskLogonParametersViaUI
			End Get
		End Property
	End Class
End Namespace
